examples/runners/wgpu/build.rs

```diff
@@ -2,7 +2,7 @@ use spirv_builder::SpirvBuilder;
 use std::error::Error;
 
 fn build_shader(path_to_create: &str) -> Result<(), Box<dyn Error>> {
-    SpirvBuilder::new(path_to_create, "spirv-unknown-vulkan1.0").build()?;
+    SpirvBuilder::new(path_to_create, "spirv-unknown-vulkan1.1").build()?;
```

Comment:
It's odd to me that this requires Vulkan 1.1. Playing with it, it seems it needs the `VK_KHR_storage_buffer_storage_class` extension, probably because the SPV refers to StorageBuffer. If you look at the [GLSL version](https://github.com/linebender/piet-gpu/blob/master/piet-gpu-hal/examples/shader/collatz.comp) and the [spv it compiles to](https://github.com/linebender/piet-gpu/blob/master/piet-gpu-hal/examples/shader/collatz.spv), it looks pretty different - it's a BufferBlock.

I'm not super expert on this, but it seems like it's using more advanced features than it really needs to, which might hurt compatibility.

Reply:
I forget if we documented this anywhere, but a decision made in rust-gpu is to be pretty opinionated in that we should only support non-deprecated SPIR-V features, and be pretty "modern" in that sense (_for now_ at least - perhaps when we have more fundamentals in place, we can look at supporting older things). `BufferBlock` is explicitly deprecated, replaced with `Block`/`StorageBuffer`, and so we don't plan on supporting `BufferBlock` right now, and so require vulkan 1.1+ for this.

Reply:
@khyperia Good to know thanks! I've been spending a lot of time looking at compatibility recently because I'm trying to target [phones,](https://developer.android.com/about/dashboards#Vulkan) and very much appreciate the load of dealing with older stuff. Having a clean, well documented line is a good strategy.